packages/cisco_ios/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -211,6 +211,14 @@ processors:
       tag: dissect_logout
       pattern: "User %{source.user.name} has %{cisco.ios.action} %{cisco.ios.session.type} session %{cisco.ios.session.number}(%{source.address})"
       if: "ctx.event?.code == 'LOGOUT'"
+  - grok:
+      field: message
+      tag: grok_badauth
+      patterns:
+        - '^No MD5 digest from %{DATA:source.address}(\(%{INT:source.port}\)|\:%{INT:source.port}) to %{DATA:destination.address}(\(%{INT:destination.port}\)|\:%{INT:destination.port}) \(RST\) ((tableid - %{DATA:cisco.ios.tableid})|(%{GREEDYDATA:_temp_.rst}))$'
+        - '^No MD5 digest from %{DATA:source.address}(\(%{INT:source.port}\)|\:%{INT:source.port}) to %{DATA:destination.address}(\(%{INT:destination.port}\)|\:%{INT:destination.port})$'
+      ignore_missing: true
+      if: ctx.event?.code == 'BADAUTH'
```

```suggestion
  - grok:
      field: message
      tag: grok_badauth
      patterns:
        - '^No MD5 digest from %{DATA:source.address}(\(%{INT:source.port}\)|\:%{INT:source.port}) to %{DATA:destination.address}(\(%{INT:destination.port}\)|\:%{INT:destination.port})(?: \(RST\) ((tableid - %{DATA:cisco.ios.tableid})|(%{GREEDYDATA:_temp_.rst})))?$'
      ignore_missing: true
      if: ctx.event?.code == 'BADAUTH'
```
?

This works and prevents the machine walking the prefix twice.

---

I thought of merging 2 patterns together later and totally forgot about that. I will do this.

---

updated as suggested